Kernel/Devices/KeyboardDevice.cpp

```diff
@@ -397,3 +415,16 @@ ssize_t KeyboardDevice::write(FileDescription&, const u8*, ssize_t)
 KeyboardClient::~KeyboardClient()
 {
 }
+
+void KeyboardDevice::set_maps(char* n_map, char* n_shift_map, char* n_alt_map)
+{
+    map = (char*) kmalloc(0x80);
+    shift_map = (char*) kmalloc(0x80);
+    alt_map = (char*) kmalloc(0x80);
```

Comment:
These need three corresponding calls to `kfree()` somewhere, otherwise repeatedly setting a keymap will leak memory every time.